### PR TITLE
Don't add empty content holder when there is no content available

### LINKED ISSUE #29127: empty h3 in popover when title is not defined
**Operating system**
Windows
**Browser and version**
Chrome 73
**Reduced test case**
https://getbootstrap.com/docs/4.3/components/popovers/#four-directions

On popovers that do not have a title defined, the h3 is still there as empty. I feel this could be an accessibility problem having an empty h3 on the page : 
image

I would make sure the h3 disappears if the title is not defined.

### PR DESCRIPTION
**Pull request by rohit2sharma95**
Fixes #29127
Alternative to #29570

A popover should not have an empty header or body when the content is not available on the main element or in the configuration.

### CODE REVIEW

File: js/src/popover.js [hunk 1]
```diff
@@ -90,6 +90,29 @@ class Popover extends Tooltip {
...
 
+  getTipElement() {
+    if (this.tip) {
+      return this.tip
+    }
+
+    const element = document.createElement('div')
+    element.innerHTML = this._config.template
+
+    this.tip = element.children[0]
```
Comment (GeoSot) [inline js/src/popover.js:101]: tile here could be `super.getTipeElement()`

File: js/src/popover.js [hunk 2]
```diff
@@ -90,6 +90,29 @@ class Popover extends Tooltip {
...
+    const title = this.getTitle()
+    const content = this._getContent()
+
+    if (!title) {
```
Comment (GeoSot) [inline js/src/popover.js:105]: you can use the method inline I suppose

File: js/src/popover.js [hunk 3]
```diff
@@ -90,6 +90,29 @@ class Popover extends Tooltip {
     return this.getTitle() || this._getContent()
   }
 
+  getTipElement() {
+    if (this.tip) {
+      return this.tip
+    }
+
+    const element = document.createElement('div')
+    element.innerHTML = this._config.template
```
Comment (XhmikosR) [inline js/src/popover.js:99]: This should respect the sanitizer like we do elsewhere.

Comment (rohit2sharma95) [reply js/src/popover.js:99]: Yes that is done in the tooltip class in the `_getConfig` method :thinking:

File: js/src/popover.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ const Default = {
   content: '',
   template: '<div class="popover" role="tooltip">' +
               '<div class="popover-arrow"></div>' +
-                '<h3 class="popover-header"></h3>' +
+              '<h3 class="popover-header"></h3>' +
```
Comment (rohit2sharma95) [inline js/src/popover.js]: @XhmikosR, I removed the space intentionally since h3 is not a nested element here :slightly_smiling_face:

Comment (XhmikosR) [reply js/src/popover.js]: Oops, sorry, let me revert it

File: js/src/popover.js (hunk 2 above)
Comment (XhmikosR) [reply js/src/popover.js:105]: The minifier should take care of it FYI

**Review by GeoSot — APPROVED** [review]

**Pull request merged**